gpkit/constraints/single_equation.py

```diff
@@ -0,0 +1,41 @@
+"Implements SingleEquationConstraint"
+from operator import le, ge, eq
+from ..small_scripts import try_str_without
+from ..repr_conventions import _str, _repr, _repr_latex_
+
+
+class SingleEquationConstraint(object):
+    "Constraint expressible in a single equation."
+    latex_opers = {"<=": "\\leq", ">=": "\\geq", "=": "="}
+    func_opers = {"<=": le, ">=": ge, "=": eq}
+
+    __str__ = _str
+    __repr__ = _repr
+    _repr_latex_ = _repr_latex_
+
+    def str_without(self, excluded=None):
+        if excluded is None:
+            excluded = ["units"]
+        return "%s %s %s" % (try_str_without(self.left, excluded),
+                             self.oper,
+                             try_str_without(self.right, excluded))
```

Comment:
hmm, no `__init__` and no self.left, self.right attributes -- I assume this class only gets inherited from? Should it be an abstract base class?